packages/cli/src/index.ts

```diff
@@ -179,6 +185,107 @@ export const initialize = async () => {
 
     await delay(2000);
 
+    console.log(
+      `Report is being served at the URL: http://localhost:9000?dir=${encodeURIComponent(
+        prefix
+      )}&type=sitemap`
+    );
+  } else if (sitemapPath) {
+    const spinnies = new Spinnies();
+
+    spinnies.add('sitemap-spinner', {
+      text: 'Parsing Sitemap',
+    });
+    let xmlString;
+
+    try {
+      xmlString = await readFile(sitemapPath, 'utf-8');
+    } catch (error) {
+      console.error('Error parsing the file');
+    }
+
+    let data;
+
+    try {
+      data = await parseStringPromise(xmlString as string);
+    } catch (error) {
+      console.error('Error parsing the file');
+      process.exit(1);
+    }
+
+    const isSiteIndex = Boolean(data['siteIndex']);
+    const isSiteMap = Boolean(data['urlset']);
+
+    if (isSiteIndex) {
+      console.log(
+        'Sorry, XML Schema for Sitemap index files is not supported by the tool'
+      );
+      process.exit(1);
+    }
+
+    if (!isSiteMap) {
+      console.log('Provided file is not a valid sitemap');
+      process.exit(1);
+    }
+
+    const urls: string[] = data['urlset']['url'].reduce(
+      (acc: string[], { loc }: { loc: string }) => acc.concat(loc),
+      []
+    );
+
+    spinnies.succeed('sitemap-spinner', {
+      text: 'Done parsing Sitemap',
+    });
+
+    const prefix = Utility.generatePrefix([...urls].shift() ?? 'untitled');
+    const directory = `./out/${prefix}`;
+    const userInput: any = await Utility.askUserInput(
+      `Provided sitemap has ${urls.length} pages. Please enter the number of pages you want to analyze (Default ${urls.length}):`,
+      { default: urls.length.toString() }
+    );
+    let numberOfUrls: number = isNaN(userInput)
+      ? urls.length
+      : parseInt(userInput);
+
+    numberOfUrls = numberOfUrls < urls.length ? numberOfUrls : urls.length;
+
+    const urlsToProcess = urls.splice(0, numberOfUrls);
+
+    const cookieAnalysisData = await analyzeCookiesUrlsInBatches(
+      urlsToProcess,
+      isHeadless,
+      DELAY_TIME,
+      cookieDictionary
+    );
+
+    spinnies.add('technology-spinner', {
```

Comment:
If we have an overall spinner for technology, can we consider the same when cookies are analyzed - [Line 254](https://github.com/GoogleChromeLabs/ps-analysis-tool/pull/271/files#diff-2fb448d11702324cc59d657d5945d59cd325540eb87b71f2e66a5bc01ada834cR254)?